diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter3/LocalVariablesWarnTest.kt

```diff
@@ -0,0 +1,535 @@
+package org.cqfn.diktat.ruleset.chapter3
+
+import com.pinterest.ktlint.core.LintError
+import generated.WarningNames
+import org.cqfn.diktat.ruleset.constants.Warnings.LOCAL_VARIABLE_EARLY_DECLARATION
+import org.cqfn.diktat.ruleset.rules.DIKTAT_RULE_SET_ID
+import org.cqfn.diktat.ruleset.rules.identifiers.LocalVariablesRule
+import org.cqfn.diktat.util.lintMethod
+import org.junit.jupiter.api.Disabled
+import org.junit.jupiter.api.Tag
+import org.junit.jupiter.api.Test
+
+class LocalVariablesWarnTest {
+    private val ruleId = "$DIKTAT_RULE_SET_ID:local-variables"
+
+    private fun warnMessage(name: String, declared: Int, used: Int) = "$name is declared on line $declared and used for the first time on line $used"
+
+    @Test
+    @Tag(WarningNames.LOCAL_VARIABLE_EARLY_DECLARATION)
+    fun `should not check top-level and member properties`() {
+        lintMethod(LocalVariablesRule(),
+                """
+                    |const val foo = 0
+                    |
+                    |class Example {
+                    |    val bar = 0
+                    |}
+                """.trimMargin()
+        )
+    }
+
+    @Test
+    @Tag(WarningNames.LOCAL_VARIABLE_EARLY_DECLARATION)
+    fun `local variables used only in this scope - positive example`() {
+        lintMethod(LocalVariablesRule(),
+                """
+                    |class Example {
+                    |    fun foo() {
+                    |        val bar = 0
+                    |        baz(bar)
+                    |        println()
+                    |    }
+                    |}
+                """.trimMargin()
+        )
+    }
+
+    @Test
+    @Tag(WarningNames.LOCAL_VARIABLE_EARLY_DECLARATION)
+    fun `local variables used only in this scope - positive example with blank lines`() {
+        lintMethod(LocalVariablesRule(),
+                """
+                    |class Example {
+                    |    fun foo() {
+                    |        val bar = 0
+                    |        
+                    |        baz(bar)
+                    |        println()
+                    |    }
+                    |}
+                """.trimMargin()
+        )
+    }
+
+    @Test
+    @Tag(WarningNames.LOCAL_VARIABLE_EARLY_DECLARATION)
+    fun `local variables used only in this scope - positive example with comments`() {
```

Comment:
need to add test that has a property in a class and same variable name in function and lambda. Something like this:
```
class E {
val a = ""
fun foo() {
val a = ""
listOf().foreach { a -> println(a) }
```